pkg/resources/pciNetDevice.go

```diff
@@ -60,6 +61,15 @@ func NewPciNetDevice(pciDevice *ghw.PCIDevice, rFactory types.ResourceFactory) (
 		Health: pluginapi.Healthy,
 	}
 
+	linkType := ""
+	if len(ifName) > 0 {
```

Comment:
Also need to check if ifName is not empty("")

Reply:
If ifName is empty then its length would be zero.

https://play.golang.org/p/cDCkPWTJydn

Reply:
duh! For some reason I was thinking ifName was a slice of names(we had that in sriov-cni where a VF could have more than one interface name)

Reply:
:)